src/LevelLoader.cpp

```diff
@@ -296,3 +296,19 @@ ActionComponent* LevelLoader::getActionComponent(json obj) {
 
    return nullptr;
 }
+
+//TODO(nurgan) find a better method for parsing to enum
+GameObjectType LevelLoader::stringToType(std::string type) {
```

Comment:
Yeah, there needs to be something better in the future.
But I couldn't find a way to directly load enums with the json libray, and besides some ugly #DEFINE hacks, there seems to no better way to parse string to enum in C++. 
So I think it is sufficient for now.